core/server_test.go

```diff
@@ -2,16 +2,24 @@ package core
 
 import (
 	"context"
+	"strings"
 	"testing"
 
+	"github.com/sirupsen/logrus/hooks/test"
 	"github.com/stretchr/testify/require"
 )
 
 func TestServerStartWithoutHandlerPanic(t *testing.T) {
+	logger, hook := test.NewNullLogger()
+	logger.SetReportCaller(true)
 	serverError := make(chan error)
 
 	require.Panics(t, func() {
-		err := Server(context.Background(), "test", nil, nil, nil, nil)
+		err := Server(context.Background(), "test", nil, nil, logger, nil)
```

Comment:
`Server` is deprecated. While you're in the area, can you refactor this to use `ServerParams`.